### ISSUE TITLE
diagnostics from interpreter don't format how I expect

### ISSUE THREAD

**Issue by nikomatsakis** [dada-lang/dada#10]
we are plumbing spans through, but when eg I run this

```
cargo run -- run dada_tests/interpret/print_wrong_label.dada 
```

I see

```
Error: Error: expected to find an argument named `message`, but found the name `greeting`


Location:
    components/dada-execute/src/error.rs:8:27
```

but I expected a nice Ariadne error with source information. What gives?

**Comment by nikomatsakis** [comment]
This is where the `eyre::Report` gets generated:

https://github.com/dada-lang/dada/blob/3e31f4aac2dd762e764ee8c41e84e5ae5d6aee48/components/dada-execute/src/error.rs#L5-L14

And here is the `Display` impl;

https://github.com/dada-lang/dada/blob/3e31f4aac2dd762e764ee8c41e84e5ae5d6aee48/components/dada-execute/src/error.rs#L24-L28

I probably just don't understand something about `eyre` and how it handles multiline reports, but ...

**Comment by f01dab1e** [comment]
`add_primary_label` has to be true?

https://github.com/dada-lang/dada/blob/aed00d805fc9e6d1818b74322bd3ad0592c2520a/components/dada-ir/src/diagnostic.rs#L115-L118

https://github.com/dada-lang/dada/blob/aed00d805fc9e6d1818b74322bd3ad0592c2520a/components/dada-ir/src/diagnostic.rs#L129

**Comment by nikomatsakis** [comment]
I found the problem, fixed in #23
